packages/syft/src/syft/node/node.py

```diff
@@ -457,49 +460,101 @@ def root_client(self):
         root_client.api.refresh_api_callback()
         return root_client
 
-    def _find_pending_migrations(self):
-        klasses_to_be_migrated = []
-
+    def _find_klasses_pending_for_migration(
+        self, object_types: List[SyftObject]
+    ) -> List[SyftObject]:
         context = AuthedServiceContext(
             node=self,
             credentials=self.verify_key,
             role=ServiceRole.ADMIN,
         )
         migration_state_service = self.get_service(MigrateStateService)
 
-        canonical_name_version_map = []
-
-        # Track all object types from document store
-        for partition in self.document_store.partitions.values():
-            object_type = partition.settings.object_type
-            canonical_name = object_type.__canonical_name__
-            object_version = object_type.__version__
-            canonical_name_version_map.append((canonical_name, object_version))
+        klasses_to_be_migrated = []
 
-        # Track all object types from action store
-        action_object_types = [Action, ActionObject]
-        action_object_types.extend(ActionObject.__subclasses__())
-        for object_type in action_object_types:
+        for object_type in object_types:
             canonical_name = object_type.__canonical_name__
             object_version = object_type.__version__
-            canonical_name_version_map.append((canonical_name, object_version))
 
-        for canonical_name, current_version in canonical_name_version_map:
             migration_state = migration_state_service.get_state(context, canonical_name)
             if (
                 migration_state is not None
                 and migration_state.current_version != migration_state.latest_version
             ):
-                klasses_to_be_migrated.append(canonical_name)
+                klasses_to_be_migrated.append(object_type)
             else:
                 migration_state_service.register_migration_state(
                     context,
-                    current_version=current_version,
+                    current_version=object_version,
                     canonical_name=canonical_name,
                 )
 
         return klasses_to_be_migrated
 
+    def find_and_migrate_data(self):
```

Comment:
Finds the objects that need migration in both Action and Document Store. Migrates the object for each Action and Document Store.